File: backend/src/services/logger.ts
const COLORS = {
  reset: '\x1b[0m',
  dim: '\x1b[2m',
  red: '\x1b[31m',
  green: '\x1b[32m',
  yellow: '\x1b[33m',
  blue: '\x1b[34m',
  magenta: '\x1b[35m',
  cyan: '\x1b[36m',
  gray: '\x1b[90m',
}

function ts() {
  return new Date().toLocaleTimeString('zh-CN', { hour12: false })
}

function log(level: string, color: string, tag: string, msg: string, data?: Record<string, any>) {
  const prefix = `${COLORS.gray}${ts()}${COLORS.reset} ${color}${level}${COLORS.reset} ${COLORS.cyan}[${tag}]${COLORS.reset}`
  const extra = data ? ` ${COLORS.dim}${JSON.stringify(data)}${COLORS.reset}` : ''
  console.log(`${prefix} ${msg}${extra}`)
}

export const logger = {
  info: (tag: string, msg: string, data?: Record<string, any>) =>
    log('INFO ', COLORS.green, tag, msg, data),

  warn: (tag: string, msg: string, data?: Record<string, any>) =>
    log('WARN ', COLORS.yellow, tag, msg, data),

  error: (tag: string, msg: string, data?: Record<string, any>) =>
    log('ERROR', COLORS.red, tag, msg, data),

  // Agent-specific logging
  agent: {
    chat: (userId: string, sessionId: string, message: string) =>
      log('CHAT ', COLORS.blue, 'agent', `user[${userId.slice(0, 8)}] session[${sessionId.slice(0, 8)}]`, { message: message.slice(0, 200) }),

    intent: (category: string, confidence: number, input: string) =>
      log('INTENT', COLORS.yellow, 'agent', `${category} (${confidence})`, { input: input.slice(0, 100) }),

    toolCall: (toolName: string, args?: any) =>
      log('TOOL ', COLORS.magenta, 'agent', `→ ${toolName}`, args ? { args: JSON.stringify(args).slice(0, 200) } : undefined),

    toolResult: (toolName: string, output: any) =>
      log('TOOL ', COLORS.magenta, 'agent', `← ${toolName}`, { output: JSON.stringify(output).slice(0, 200) }),

    streamStart: (sessionId: string) =>
      log('STREAM', COLORS.cyan, 'agent', `streaming started [${sessionId.slice(0, 8)}]`),

    streamEnd: (sessionId: string, textLength: number) =>
      log('STREAM', COLORS.cyan, 'agent', `streaming ended [${sessionId.slice(0, 8)}]`, { chars: textLength }),

    blocked: (category: string, input: string) =>
      log('BLOCK', COLORS.red, 'agent', `blocked ${category}`, { input: input.slice(0, 100) }),

    memory: (action: string, content: string) =>
      log('MEMORY', COLORS.green, 'agent', `${action}`, { content: content.slice(0, 100) }),

    tripCreated: (tripId: string, title: string) =>
      log('TRIP ', COLORS.green, 'agent', `created`, { tripId: tripId.slice(0, 8), title }),

    tripModified: (tripId: string, action: string) =>
      log('TRIP ', COLORS.yellow, 'agent', `modified: ${action}`, { tripId: tripId.slice(0, 8) }),
  },
}

File: backend-py/app/logger.py
from __future__ import annotations

import json
from datetime import datetime
from typing import Any, Optional


class _C:
    reset = "\x1b[0m"
    dim = "\x1b[2m"
    red = "\x1b[31m"
    green = "\x1b[32m"
    yellow = "\x1b[33m"
    blue = "\x1b[34m"
    magenta = "\x1b[35m"
    cyan = "\x1b[36m"
    gray = "\x1b[90m"


def _ts() -> str:
    return datetime.now().strftime("%H:%M:%S")


def _log(level: str, color: str, tag: str, msg: str, data: Optional[dict] = None) -> None:
    prefix = f"{_C.gray}{_ts()}{_C.reset} {color}{level}{_C.reset} {_C.cyan}[{tag}]{_C.reset}"
    extra = f" {_C.dim}{json.dumps(data, ensure_ascii=False)}{_C.reset}" if data else ""
    print(f"{prefix} {msg}{extra}")


class _AgentLogger:
    @staticmethod
    def chat(user_id: str, session_id: str, message: str) -> None:
        _log("CHAT ", _C.blue, "agent", f"user[{user_id[:8]}] session[{session_id[:8]}]",
             {"message": message[:200]})

    @staticmethod
    def intent(category: str, confidence: float, input_str: str) -> None:
        _log("INTENT", _C.yellow, "agent", f"{category} ({confidence})",
             {"input": input_str[:100]})

    @staticmethod
    def tool_call(tool_name: str, args: Optional[Any] = None) -> None:
        _log("TOOL ", _C.magenta, "agent", f"\u2192 {tool_name}",
             {"args": json.dumps(args, ensure_ascii=False)[:200]} if args else None)

    @staticmethod
    def tool_result(tool_name: str, output: Any) -> None:
        _log("TOOL ", _C.magenta, "agent", f"\u2190 {tool_name}",
             {"output": json.dumps(output, ensure_ascii=False)[:200]})

    @staticmethod
    def stream_start(session_id: str) -> None:
        _log("STREAM", _C.cyan, "agent", f"streaming started [{session_id[:8]}]")

    @staticmethod
    def stream_end(session_id: str, text_length: int) -> None:
        _log("STREAM", _C.cyan, "agent", f"streaming ended [{session_id[:8]}]",
             {"chars": text_length})

    @staticmethod
    def blocked(category: str, input_str: str) -> None:
        _log("BLOCK", _C.red, "agent", f"blocked {category}",
             {"input": input_str[:100]})

    @staticmethod
    def memory(action: str, content: str) -> None:
        _log("MEMORY", _C.green, "agent", action,
             {"content": content[:100]})

    @staticmethod
    def trip_created(trip_id: str, title: str) -> None:
        _log("TRIP ", _C.green, "agent", "created",
             {"tripId": trip_id[:8], "title": title})

    @staticmethod
    def trip_modified(trip_id: str, action: str) -> None:
        _log("TRIP ", _C.yellow, "agent", f"modified: {action}",
             {"tripId": trip_id[:8]})


class Logger:
    @staticmethod
    def info(tag: str, msg: str, data: Optional[dict] = None) -> None:
        _log("INFO ", _C.green, tag, msg, data)

    @staticmethod
    def warn(tag: str, msg: str, data: Optional[dict] = None) -> None:
        _log("WARN ", _C.yellow, tag, msg, data)

    @staticmethod
    def error(tag: str, msg: str, data: Optional[dict] = None) -> None:
        _log("ERROR", _C.red, tag, msg, data)

    agent = _AgentLogger()


logger = Logger()
